File: Extasys for .NET/Extasys.Examples.TCPClient/TCPClient.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading;
using Extasys.Network.TCP.Client.Exceptions;

namespace Extasys.Examples.TCPClient
{
    public class TCPClient:Extasys.Network.TCP.Client.ExtasysTCPClient
    {
        private Thread fAutoSendMessages;
        public TCPClient(string name,string description)
            :base(name,description)
        {
        }

        public override void OnDataReceive(Extasys.Network.TCP.Client.Connectors.TCPConnector connector, Extasys.Network.DataFrame data)
        {
            string dataReceived = Encoding.ASCII.GetString(data.Bytes);

        }

        public override void OnConnect(Extasys.Network.TCP.Client.Connectors.TCPConnector connector)
        {
            Console.WriteLine("Connected to server");
        }

        public override void OnDisconnect(Extasys.Network.TCP.Client.Connectors.TCPConnector connector)
        {
            Console.WriteLine("Disconnected from server");
        }

        public void StartSendingMessages()
        {
            if (fAutoSendMessages == null)
            {
                fAutoSendMessages = new Thread(new ThreadStart(AutoSendMessages));
                fAutoSendMessages.Start();
            }
            else
            {
                fAutoSendMessages.Abort();
                fAutoSendMessages = new Thread(new ThreadStart(AutoSendMessages));
                fAutoSendMessages.Start();
            }
        }

        public void StopSendingMessages()
        {
            try
            {
                fAutoSendMessages.Abort();
            }
            catch (Exception ex)
            {
            }
        }

        private void AutoSendMessages()
        {
            int messageCount = 0;
            while (true)
            {
                try
                {
                    messageCount++;
                    SendData(messageCount.ToString() + ((char)2)); // Char 2 is the message splitter the server's message collector uses.
                    Thread.Sleep(2);
                }
                catch (ConnectorDisconnectedException ex)
                {
                    Console.WriteLine(ex.ConnectorInstance.Name + " connector disconnected!");
                    StopSendingMessages();
                }
                catch (ConnectorCannotSendPacketException ex)
                {
                    Console.WriteLine("Connector " + ex.ConnectorInstance.Name + " cannot send packet");
                    StopSendingMessages();
                }
                catch (Exception ex)
                {
                   StopSendingMessages();
                }
            }
        }


    }
}

File: Extasys for Java Examples/Extasys.Examples.TCPClient/src/Extasys/Examples/TCPClient/TCPClient.java
/*Copyright (c) 2008 Nikos Siatras

 Permission is hereby granted, free of charge, to any person obtaining a copy
 of this software and associated documentation files (the "Software"), to deal
 in the Software without restriction, including without limitation the rights
 to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
 copies of the Software, and to permit persons to whom the Software is
 furnished to do so, subject to the following conditions:

 The above copyright notice and this permission notice shall be included in
 all copies or substantial portions of the Software.

 THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
 IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
 FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
 AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
 LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
 OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
 THE SOFTWARE.*/
package Extasys.Examples.TCPClient;

import Extasys.DataFrame;
import Extasys.DataConvertion.Base64Converter;
import Extasys.Network.TCP.Client.Connectors.TCPConnector;
import Extasys.Network.TCP.Client.Exceptions.ConnectorCannotSendPacketException;
import Extasys.Network.TCP.Client.Exceptions.ConnectorDisconnectedException;
import java.net.InetAddress;

/**
 *
 * @author Nikos Siatras
 */
public class TCPClient extends Extasys.Network.TCP.Client.ExtasysTCPClient
{

    private boolean fKeepSendingMessages = false;
    private int fPreviousNumber = 0;

    public TCPClient(String name, String description, InetAddress remoteHostIP, int remoteHostPort, int corePoolSize, int maximumPoolSize)
    {
        super(name, description, corePoolSize, maximumPoolSize);

        try
        {
            // Add a new connector to this TCP Client
            TCPConnector connector = super.AddConnector(name, remoteHostIP, remoteHostPort, 8192, ((char) 3));
            connector.setAutoApplyMessageSplitterState(true); // Auto apply message splitter to outgoing messages
            connector.setConnectionDataConverter(new Base64Converter()); // Base 64 Encoding
        }
        catch (Exception ex)
        {
        }
    }

    @Override
    public void OnDataReceive(TCPConnector connector, DataFrame data)
    {
        try
        {
            final String dataReceivedFromServer = new String(data.getBytes());

            int number = Integer.parseInt(dataReceivedFromServer);
            if (fPreviousNumber != number)
            {
                System.err.println("Message Lost");
            }
            else
            {
                fPreviousNumber += 1;
            }

            // Every time I receive data from the server I send the
            // fMessageToExchange string back
            if (fKeepSendingMessages)
            {
                connector.SendData(String.valueOf(fPreviousNumber));
            }
        }
        catch (ConnectorCannotSendPacketException | ConnectorDisconnectedException ex)
        {
            System.err.println(ex.getMessage());
        }
    }

    @Override
    public void OnConnect(TCPConnector connector)
    {
        System.out.println("Connected to server");
    }

    @Override
    public void OnDisconnect(TCPConnector connector)
    {
        System.out.println("Disconnected from server");
        StopSendingMessages();
    }

    public void StartSendingMessages()
    {
        fKeepSendingMessages = true;
        try
        {
            fPreviousNumber = 1000000000;
            SendData(String.valueOf(fPreviousNumber));
        }
        catch (ConnectorDisconnectedException | ConnectorCannotSendPacketException ex)
        {
            fKeepSendingMessages = false;
        }
    }

    public void StopSendingMessages()
    {
        fKeepSendingMessages = false;
    }
}
